### ISSUE TITLE
`RUF022` does not follow `lint.isort.classes` option

### ISSUE THREAD

**Issue by velsinki** [astral-sh/ruff#20362]
### Summary

The following example

```python
"""Example __init__.py problem."""
from .defs import AnotherClass, CLASS

__all__ = [
    "AnotherClass",
    "CLASS",
]
```

with this option in `ruff.toml`

```toml
[lint.isort]
classes = ["CLASS"]
```

should result in no errors because of the overridden PEP8 type inference with `classes`, but `RUF022` still triggers.

This is a small issue which I only noticed because the order of `__all__` can be used in `sphinx-autodoc`. Still, this seems unexpected behavior in my opinion.

**Comment by ntBre** [comment]
This makes some sense to me, my only hesitation is that we we have a lot of isort settings, and it doesn't look like the rule accounts for any of them, preferring a simpler sort. I think that's in line with the "isort-style" part of the docs. It could be a bit of a slippery slope to start using some of the isort options, but I'm open to other thoughts!

This is more of an implementation detail than an essential issue, but structurally it also looks like it would be difficult to access the `LinterSettings` where the sort actually takes place.

**Comment by velsinki** [comment]
I see, thanks for responding so quickly :)

I would understand why this is hard to fix or out of scope, but maybe this can somehow be documented? For example `RUF022` could state that (currently?) the `isort` settings are ignored?

**Comment by ntBre** [comment]
Ah that's a good idea to clarify this in the rule docs!
